homeassistant/components/zwave_js/__init__.py

```diff
@@ -190,21 +199,42 @@ def async_on_value_notification(notification: ValueNotification) -> None:
         )
 
     @callback
-    def async_on_notification(notification: Notification) -> None:
+    def async_on_notification(
+        notification: EntryControlNotification | NotificationNotification,
+    ) -> None:
         """Relay stateless notification events from Z-Wave nodes to hass."""
         device = dev_reg.async_get_device({get_device_id(client, notification.node)})
-        hass.bus.async_fire(
-            ZWAVE_JS_EVENT,
-            {
-                ATTR_TYPE: "notification",
-                ATTR_DOMAIN: DOMAIN,
-                ATTR_NODE_ID: notification.node.node_id,
-                ATTR_HOME_ID: client.driver.controller.home_id,
-                ATTR_DEVICE_ID: device.id,  # type: ignore
-                ATTR_LABEL: notification.notification_label,
-                ATTR_PARAMETERS: notification.parameters,
-            },
-        )
+        event_data = {
+            ATTR_TYPE: "notification",
+            ATTR_DOMAIN: DOMAIN,
+            ATTR_NODE_ID: notification.node.node_id,
+            ATTR_HOME_ID: client.driver.controller.home_id,
+            ATTR_DEVICE_ID: device.id,  # type: ignore
+            ATTR_COMMAND_CLASS: notification.command_class,
+        }
+
+        if isinstance(notification, EntryControlNotification):
+            event_data.update(
+                {
+                    ATTR_COMMAND_CLASS_NAME: "Entry Control",
+                    ATTR_EVENT_TYPE: notification.event_type,
+                    ATTR_DATA_TYPE: notification.data_type,
+                    ATTR_EVENT_DATA: notification.event_data,
+                }
+            )
+        else:
+            event_data.update(
+                {
+                    ATTR_COMMAND_CLASS_NAME: "Notification",
+                    ATTR_LABEL: notification.label,
+                    ATTR_TYPE_: notification.type_,
```

Comment:
I think we should call the item `"type"` if that matches the zwave-js item name.